client/human/HumanClientFSM.cpp

```diff
@@ -189,16 +205,24 @@ WaitingForMPJoinAck::~WaitingForMPJoinAck()
 boost::statechart::result WaitingForMPJoinAck::react(const JoinGame& msg) {
     if (TRACE_EXECUTION) DebugLogger() << "(HumanClientFSM) WaitingForMPJoinAck.JoinGame";
 
-    Client().Networking().SetPlayerID(msg.m_message.ReceivingPlayer());
+    try {
+        int player_id = boost::lexical_cast<int>(msg.m_message.Text());
+
+        Client().Networking().SetPlayerID(player_id);
 
-    return transit<MPLobby>();
+        return transit<MPLobby>();
+    } catch (...) {
```

Comment:
Again, just catch the `bad_lexical_cast` or add to the serialization of this message type.